src/jlgen.jl

```diff
@@ -518,7 +542,7 @@ end
 function ci_cache_lookup(cache, mi, min_world, max_world)
     wvc = WorldView(cache, min_world, max_world)
     ci = CC.get(wvc, mi, nothing)
-    if ci !== nothing && ci.inferred === nothing
+    if VERSION < v"1.12.0-DEV.1434" && ci !== nothing && ci.inferred === nothing
```

Comment:
I'm not sure about this; we encounter CIs with `inferred` set to nothing, and those seem safe to pass to codegen now?

Reply:
Normally most CI should have their value set to `nothing`. Anything else you might have observed (whether on this version or in the past) is a transient state and can change concurrently, so is unsafe to rely upon

Reply:
So you're saying that check is not necessary? If I don't forcibly populate our CI cache for CIs with `ci.inferred == nothing`, codegen calls into inference which is disallowed (`Refusing to automatically run type inference with custom cache lookup.`). On 1.11, at least.